src/python/pants/jvm/target_types.py

```diff
@@ -154,6 +154,21 @@ class JvmProvidesTypesField(StringSequenceField):
     )
 
 
+class JvmArtifactCompatibleResolvesField(JvmCompatibleResolvesField):
+    help = (
```

Comment:
Writing is hard...feedback welcomed for both these new help messages

Reply:
So, it sortof feels like rather than customizing the help for use with `JvmArtifact`, you could expand the help for `JvmCompatibleResolvesField` to indicate that it applies across all targets, and defines a universe of code that uses a particular thirdparty resolve... i.e., move some of the help you've added here onto the existing field instead.

Reply:
I still need to update `JvmCompatibleResolvesField` in a followup, but I do think it's valuable for these to diverge. In particular, the behavior is different for how we validate that the entire transitive closure is compatible with each other. For `jvm_artifact`, it by definition is the leaf(?) of the closure as it does not have the `dependencies` field. So it doesn't make sense to talk about validating its dependencies.

Reply:
Hm. Maybe. But I think that similar to `interpreter_constraints`, you need to understand the big picture before you can successfully set the field anywhere. Splitting the big picture up across multiple fields makes that a bit harder. But there might be a few details that are only relevant here.

Reply:
Yeah I think we should do a comprehensive audit of `help` messages for JVM at the same time we write the docs for it. (I like writing docs, so I can do that when the time is right if you'd like)